sdk/storage/inc/blobs/protocol/blob_rest_client.hpp

```diff
@@ -1039,30 +1039,19 @@ namespace Azure { namespace Storage { namespace Blobs {
           Azure::Core::Context context,
           std::unique_ptr<Azure::Core::Http::RawResponse> pHttpResponse)
       {
-        unused(context);
         Azure::Core::Http::RawResponse& httpResponse = *pHttpResponse;
         ListContainersSegment response;
         auto http_status_code
             = static_cast<std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
                 httpResponse.GetStatusCode());
         if (!(http_status_code == 200))
         {
-          throw StorageError::CreateFromResponse(std::move(pHttpResponse));
+          throw StorageError::CreateFromResponse(context, std::move(pHttpResponse));
         }
         {
-          auto bodyStream = httpResponse.GetBodyStream();
-          std::vector<uint8_t> bodyContent;
-          if (bodyStream->Length() == -1)
-          {
-            bodyContent = Azure::Core::Http::BodyStream::ReadToEnd(context, *bodyStream);
-          }
-          else
-          {
-            bodyContent.resize(static_cast<std::size_t>(bodyStream->Length()));
-            Azure::Core::Http::BodyStream::ReadToCount(
-                context, *bodyStream, &bodyContent[0], bodyStream->Length());
-          }
-          XmlReader reader(reinterpret_cast<const char*>(bodyContent.data()), bodyContent.size());
+          const auto& httpResponseBody = httpResponse.GetBody();
```

Comment:
What if customer sets context's `Azure::Core::Http::Details::c_GetStreamForBody` to true?

Reply:
Then this whole thing won't work anymore.

Reply:
My point is if customer sets it to true, should we overwrite it with false?

Reply:
See my commit for what I meant.

Reply:
don't think it's necessary at all.

Reply:
First the customer is not supposed to set this value. If he did, maybe he wants to get the raw HTTP response body.

Reply:
The customer might be sharing context in multiple requests, and not be aware of the fact that it will break the APIs with a response body to deserialize.

Reply:
Still, the customers aren't supposed to disrupt normal operation inside this SDK. If they insist on doing it, they should take the consequences.

Reply:
I think you are right, but my fear is that this is not an obvious mistake to avoid.
I wouldn't know when I ask the SDK to return stream on responses will cause list operations to stop working if I'm the customer, there is just little clue for it.

Reply:
why would you ever ask the sdk to return stream as a customer? don't trouble yourself

Reply:
i feel that you're thinking too much for our customers, too much that no customer would ever actually do.

What if customer provides a pipeline and call `std::abort` in the pipeline, how are we going to save the program from crash? There's no way, we don't need to think about that.

Reply:
@katmsft Please remove the line that sets the value to false to unblock ourselves before that bug (if you want to call it a bug) is fixed. There're another two PRs waiting in line.

After it is fixed, you can add that line of code back if you want, although I don't think it's necessary at all.

Reply:
The transport policy is not even taking care about the value in the context option. Only having the the option with any value was enough to make transport policy to return the streamBody.  
I think there´s a point on @katmsft argument. I will talk to @JeffreyRichter about it.

Reply:
talked to @JeffreyRichter.  The conclusion is to have context options only when we know we want to expose that option for customers usage.
In this case, we don't want to expose this feature, so I will move the option to the RawRequest instead